ui/src/app/views/dataset-page/component.tsx

```diff
@@ -455,24 +534,54 @@ export const DataSetPage = withCurrentWorkspace()(class extends React.Component<
       </FadeBox>
       <FadeBox style={{marginTop: '1rem'}}>
         <div style={{backgroundColor: 'white', border: '1px solid #E5E5E5'}}>
-          <div style={{...styles.selectBoxHeader, display: 'flex', position: 'relative'}}>
-            <div>Preview Dataset</div>
-            <div style={{marginLeft: '1rem', color: '#000000', fontSize: '14px'}}>A visualization
+          <div style={{...styles.selectBoxHeader, display: 'flex', flexDirection: 'row',
+            position: 'relative'}}>
+            <div style={{color: '#000000', fontSize: '14px'}}>A visualization
               of your data table based on the variable and value you selected above</div>
             <Button style={{position: 'absolute', right: '8rem', top: '.25rem'}}
-                    onClick={() => {
-                      this.generateCode();
-                    }}>
+                    disabled={this.disableSave()} onClick={() => {this.generateCode(); }}>
               GENERATE CODE
             </Button>
-            {/* Button disabled until this functionality added*/}
+            <Button style={{position: 'absolute', right: '16rem', top: '0.25rem'}}
+                    disabled={this.disableSave()} onClick={() => {this.getPreviewData(); }}>
+              PREVIEW DATA SET
+            </Button>
             <Button style={{position: 'absolute', right: '1rem', top: '.25rem'}}
                     onClick ={() => this.setState({openSaveModal: true})}
                     disabled={this.disableSave()}>
-              SAVE DATASET
+              SAVE DATA SET
             </Button>
           </div>
-          {/*TODO: Display dataset preview*/}
+          {previewDataLoading && <div style={{display: 'flex', flexDirection: 'column'}}>
+            <Spinner style={{position: 'relative', top: '2rem',
+              left: '20rem'}}></Spinner>
```

Comment:
Rather than using a left property, can you center it and get rid of the position relative?